tests/test-prune.sh

```diff
@@ -0,0 +1,292 @@
+cat prune.log
+
+assert_file_has_content prune.log "Total objects: 61"
+
+count_objects repo
+assert_streq $NUM_FILE 26
+assert_streq $NUM_DIRTREE 23
+assert_streq $NUM_COMMIT 10
+assert_streq $NUM_DIRMETA 2
+assert_streq $NUM_COMMITMETA2 $NUM_COMMIT
+assert_streq $NUM_OBJECT 71
+
+ok "no-op prune"
+
+############# Try various depth prunes, with and without .commitmeta2
+
+rm -rf repo
+cp -ra orig-repo repo # Work on a copy
+
+# depth = 2 will only remove one commit from app2
+
+$FLATPAK build-update-repo --no-update-summary --no-update-appstream --prune --prune-depth=2 repo > prune.log
+cat prune.log
+
+assert_file_has_content prune.log "Total objects: 61"
+assert_file_has_content prune.log "Deleted 5 objects,"
+
+count_objects repo
+assert_streq $NUM_FILE 24
+assert_streq $NUM_DIRTREE 21
+assert_streq $NUM_COMMIT 9
+assert_streq $NUM_DIRMETA 2
+assert_streq $NUM_COMMITMETA2 $NUM_COMMIT
+assert_streq $NUM_OBJECT 65
+
+rm -rf repo
+cp -ra incremental-repo repo # Work on a copy w/ commitmeta2s
+
+# depth = 2 will only remove one commit from app2
+
+$FLATPAK build-update-repo --no-update-summary --no-update-appstream --prune --prune-depth=2 repo > prune.log
+cat prune.log
+
+assert_file_has_content prune.log "Total objects: 61"
+assert_file_has_content prune.log "Deleted 5 objects,"
+
+count_objects repo
+assert_streq $NUM_FILE 24
+assert_streq $NUM_DIRTREE 21
+assert_streq $NUM_COMMIT 9
+assert_streq $NUM_DIRMETA 2
+assert_streq $NUM_COMMITMETA2 $NUM_COMMIT
+assert_streq $NUM_OBJECT 65
+
+# depth = 1 will only remove 4 commits
+
+rm -rf repo
+cp -ra orig-repo repo # Work on a copy
+
+$FLATPAK build-update-repo --no-update-summary --no-update-appstream --prune --prune-depth=1 repo > prune.log
+cat prune.log
+
+assert_file_has_content prune.log "Total objects: 61"
+assert_file_has_content prune.log "Deleted 20 objects,"
+
+count_objects repo
+assert_streq $NUM_FILE 18
+assert_streq $NUM_DIRTREE 15
+assert_streq $NUM_COMMIT 6
+assert_streq $NUM_DIRMETA 2
+assert_streq $NUM_COMMITMETA2 $NUM_COMMIT
+assert_streq $NUM_OBJECT 47
+
+rm -rf repo
+cp -ra incremental-repo repo # Work on a copy w/ commitmeta2s
+
+$FLATPAK build-update-repo --no-update-summary --no-update-appstream --prune --prune-depth=1 repo > prune.log
+cat prune.log
+
+assert_file_has_content prune.log "Total objects: 61"
+assert_file_has_content prune.log "Deleted 20 objects,"
+
+count_objects repo
+assert_streq $NUM_FILE 18
+assert_streq $NUM_DIRTREE 15
+assert_streq $NUM_COMMIT 6
+assert_streq $NUM_DIRMETA 2
+assert_streq $NUM_COMMITMETA2 $NUM_COMMIT
+assert_streq $NUM_OBJECT 47
+
+ok "depth prune"
+
+############# Try non-reachable prunes, with and without .commitmeta2
+
+rm -rf repo
+cp -ra orig-repo repo # Work on a copy
+
+rm repo/refs/heads/app3 # Removes 3 commits
+
+$FLATPAK build-update-repo --no-update-summary --no-update-appstream --prune --prune-depth=-1 repo > prune.log
+cat prune.log
+assert_file_has_content prune.log "Total objects: 61"
```

Comment:
Could add a line checking the "Deleted x objects" output to this and the rest of the prunes for completeness